ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/TransactionComponents/AcceptTransaction.qml

```diff
@@ -77,24 +77,39 @@ Item {
         }
     }
 
-    SignTransactionModal {
+    Loader {
         id: signTransactionModal
-        onOpened: {
-          walletModel.getGasPricePredictions()
+        function open() {
+            this.sourceComponent = cmpSignTransactionModal
+            this.item.open()
         }
-        selectedAccount: {}
-        selectedRecipient: {
-            return {
-                address: commandParametersObject.address,
-                identicon: chatsModel.activeChannel.identicon,
-                name: chatsModel.activeChannel.name,
-                type: RecipientSelector.Type.Contact
+        function closed() {
+            this.sourceComponent = undefined // kill an opened instance
```

Comment:
FYI, you can set `active` to `false` to kill an instance of a component instead of removing the sourceComponent. It should do the same effect.

By doing that, you can also set the sourceComponent inside the Loader to save space. eg:
```qml
    Loader {
        id: signTransactionModal
        active: conditoonOrSomething
        sourceComponent:  Component {
              id: cmpSignTransactionModal
              ...
```

Reply:
Nice, I didn't realise `active` also loaded/unloaded the source component. I've updated it so that the `sourceComponent` property holds the modal, however I wasn't able to add an active condition as it's controlled by the `open()` and `close()` actions of the modal (`.opened` wouldn't work either). Great suggestion, thank you!